wqflask/base/data_set.py

```diff
@@ -1311,7 +1311,8 @@ def fetch_cached_results(dataset_name: str, dataset_type: str):
     file_path = os.path.join(TMPDIR, f"{file_name}.json")
     try:
         with open(file_path, "r") as file_handler:
+            # return json.load(file_handler)
+            return [obj for obj in ijson.items(file_handler, "", use_float=True)][0]
```

Comment:
*question:* Wouldn't it be better if we returned an iterator, instead of a list?